File: 2042-check-if-numbers-are-ascending-in-a-sentence/2042-check-if-numbers-are-ascending-in-a-sentence.py
class Solution:
    def areNumbersAscending(self, s: str) -> bool:
        length=len(s)
        number=0
        i=0
        while i<length:
            temp=''
            if s[i]>='0' and s[i]<='9':
                while i<length and s[i]!=" ":
                    temp+=s[i]
                    i+=1
                if int(temp)<=number:
                    return False
                number=int(temp)
                
            # number=int(temp) 
            if i<length:
                i+=1
            
        return True
               

File: 2042-check-if-numbers-are-ascending-in-a-sentence/2042-check-if-numbers-are-ascending-in-a-sentence.cpp
class Solution {
public:
    bool areNumbersAscending(string s) {
        int n =s.length();
        int prev=0;
        for(int i=0;i<n;i++){
            string temp="";
            if(s[i]>='1' && s[i]<='9'){
                while(i<n && s[i]!=' '){
                    temp+=s[i];
                    i++;
                }
                if(!temp.empty() && stoi(temp)<=prev)return false;
            }
            if(!temp.empty())prev=stoi(temp);
            
        }
        return true;
    }
};